Lib/argparse.py

```diff
@@ -707,12 +708,73 @@ class MetavarTypeHelpFormatter(HelpFormatter):
     """
 
     def _get_default_metavar_for_optional(self, action):
+        if action.type is None:
+            return "None"
         return action.type.__name__
 
     def _get_default_metavar_for_positional(self, action):
+        if action.type is None:
+            return "None"
         return action.type.__name__
 
 
+class _CustomizableHelpFormatter(RawTextHelpFormatter,
+                                ArgumentDefaultsHelpFormatter,
+                                MetavarTypeHelpFormatter):
+
+    def __init__(self, raw_description, raw_text,
+            arg_defaults, metavar_type,
+            **kwargs):
+        super(_CustomizableHelpFormatter, self).__init__(**kwargs)
+        if not raw_description:
+            self.disable_raw_description()
+        if not raw_text:
+            self.disable_raw_text()
+        if not arg_defaults:
+            self.disable_arg_defaults()
+        if not metavar_type:
+            self.disable_metavar_type()
+
+    def bind_sub(self, method):
+        """Bind method from super to self."""
+        setattr(self, method.__name__, method.__get__(self))
+
+    def disable_raw_description(self):
+        self.bind_sub(HelpFormatter._fill_text)
+
+    def disable_raw_text(self):
+        self.bind_sub(HelpFormatter._split_lines)
+
+    def disable_arg_defaults(self):
+        self.bind_sub(HelpFormatter._get_help_string)
+
+    def disable_metavar_type(self):
+        self.bind_sub(HelpFormatter._get_default_metavar_for_optional)
+        self.bind_sub(HelpFormatter._get_default_metavar_for_positional)
+
+
+class CustomHelpFormat:
+    """Customizable help formatter factory."""
+
+    def __init__(self):
```

Comment:
why not allow to pass the attributes by arguments ? 

```python
def __init__(self, *, 
             indent_increment: int = 2, 
             max_help_position: int = 24,
                    ...
):
        ...
```

Reply:
I wanted to avoid backward incompatiblity when list of attributes change. On the other hand, this can be solved by adding `**kwargs` at the end, so I think I'll try it this way.